### ISSUE TITLE
v2 no longer supports AWS Local Zone ALBs

### ISSUE THREAD

**Issue by clayvan** [kubernetes-sigs/aws-load-balancer-controller#1851]
We run in a Local Zone currently using ALB Ingress Controller v1, which is similar but different from AWS Outposts.

In v1 (ALB ingress controller) our ALB was able to be created by the following annotations:
```yaml
kubernetes.io/ingress.class: alb
alb.ingress.kubernetes.io/scheme: internet-facing
alb.ingress.kubernetes.io/target-type: ip
alb.ingress.kubernetes.io/certificate-arn: arn:aws:acm:REGION:ACCOUNT:certificate/UUID
alb.ingress.kubernetes.io/subnets: subnet-12345
```
Because we are using private IPs for the ALB, but it needs to be internet facing to grab a customer owned IP, we had to specify the subnets as we don't want to tag them as internetfacing. There was no requirement on needing 2 subnets, so we had no issues here.

In AWS Load Balancer Controller v2, I'm not able to deploy an ALB in Local Zone for a variety of different error messages depending on what I try.

If I use the same config as before, I get
> subnets count less than minimal required count: 1 < 2

If I don't specify the `alb.ingress.kubernetes.io/customer-owned-ipv4-pool` flag, I get
> Failed deploy model due to ValidationError: You must use a customer owned IPv4 pool with an Application Load Balancer on an Outpost

If I don't supply a subnet I get
>  ingress  Failed build model due to couldn't auto-discover subnets: unable to discover at least one subnet

If I supply 2 subnets in the same Local Zone I get
> ingress  Failed build model due to multiple subnets in same Availability Zone myLocalZoneID: [subnet-123 subnet-456]

I think what's happening is that as a part of https://github.com/kubernetes-sigs/aws-load-balancer-controller/pull/1627 and https://github.com/kubernetes-sigs/aws-load-balancer-controller/pull/1685, we added checking for AWS Outposts and support of Customer Owned IP pools, which @M00nF1sh did validate with a single subnet like I use. However Local Zones don't show an Outpost ARN.

Therefore our subnets are being treated as Public region and requires 2 subnets in 2 AZs.

If I had to guess, this was unexpected behavior. Thoughts on how to move forward so we can get onto v2?

**Comment by kishorj** [comment]
/kind feature
/assign m00nf1sh
we will be adding support for the Local Zone in the upcoming v2.2.0 release.

**Comment by kishorj** [comment]
Support available in v2.2.0 release. For details, please refer to the release notes [[link](https://github.com/kubernetes-sigs/aws-load-balancer-controller/releases/tag/v2.2.0)].
